### DISCUSSION TITLE
Keep AdminUI restcalls in scope of a Realm

### DISCUSSION THREAD

**Discussion by RaffaelGrob** [keycloak/keycloak#24199] [Archive]
I'm trying to protect each realm individually by a reverse-proxy by routing 
- /admin/realms/my-realm/ 
- /admin/my-realm 
- /realms/my-realm 
to a specific Keycloak instance. The goal is to have multiple keycloak servers behind a common naming scheme. And I want to make it possible that owners of a specific realm can use the admin-ui at: https://my.keycloak.net/admin/**my-realm**/console/

While testing this, i discovered that in the Admin Console some endpoint calls are made outside the realm-scope. Examples:
- when the realms dropdown is used ->/admin/realms?briefRepresentation=true
- and occasionally -> /admin/serverinfo/ is called
Better would be if these (I yet don't know if the list is complete) end points were made available to the Admin Console under a realm specific path like /admin/realms/**my-realm**/?briefRepresentation=true

I'm aware of other shared pathes like: /robots.txt /js/ /resources/ which are realm unspecific. They don't pose an issue assuming all Keycloak Instances behind a proxy run the same Version. However, why not to make all of them also available within the context of /realms/**my-realm**/resources?

I'm not sure if the total isolation of each realm is an objective of the keycloak project. Therefore I firstly mentioned my request at this issue: https://github.com/keycloak/keycloak/pull/20169. Later I decided to open a dedicated issue for it. Please advise if you need further info or another way to approach this. Many thanks.

**Comment by antikalk** [comment]
> when the realms dropdown is used ->/admin/realms?briefRepresentation=true

Maybe it would be a possibility to when using the Account Console in a dedicated realm scope to instead of rendering the dropdown to just simply display the realm name. Hereby the call to the problematic URL would be omitted. The user would not be able to select any other realm anyway. So: if master admin console, render the dropdown, otherwise not.

But the serverinfo is then still a problem.

**Comment by jonkoops** [comment]
Note that we also make other calls to the root of the Keycloak server. Such as the [Well-known URI](https://en.wikipedia.org/wiki/Well-known_URI).
